solidity/test/bank/Bank.test.ts

```diff
@@ -424,15 +424,33 @@ describe("Bank", () => {
       before(async () => {
         await createSnapshot()
         await bank.connect(owner).approveBalance(spender, to1e18(1337))
-        await bank.connect(owner).approveBalance(spender, amount)
       })
 
       after(async () => {
         await restoreSnapshot()
       })
 
-      it("should replace the previous allowance", async () => {
-        expect(await bank.allowance(owner.address, spender)).to.equal(amount)
+      context("when setting approval back to zero", () => {
+        before(async () => {
+          await createSnapshot()
+          await bank.connect(owner).approveBalance(spender, 0)
+        })
+
+        after(async () => {
+          await restoreSnapshot()
+        })
+
+        it("should replace the previous allowance with zero", async () => {
+          expect(await bank.allowance(owner.address, spender)).to.equal(0)
+        })
+      })
+
+      context("when trying to overwrite with a non-zero value", () => {
```

Comment:
Can we add a test case where the current approval is zero and the new approval is zero?

Reply:
Sure, on it.

Reply:
d782d5800b4da239d1346b33e6ae0c4d89cbfa5c